proposals/NNNN-package-manager-product-definitions.md

```diff
@@ -7,135 +7,123 @@
 
-As mentioned earlier, Swift Package Manager already infers a set of implicit products based on the set of targets in the package.  One open question is whether inference rules will be supported and documented as part of this feature, in order to allow existing packages to continue to work unmodified and to avoid the need for explicit product definitions for simple packages.  While the inference rules will need to be supported for existing packages, it is still an open question whether product will be inferred for packages using the Swift Pacakge Manager 4.0 version of the API.
-
-The inference rules that apply to legacy packages are detailed in the next section of this document.
-
-Even if Swift Pacakge Manager 4.0 does infer products for non-legacy packages, products will only be inferred if there are no explicit product definitions in the package.  The reason is that the interaction of a mixture of implicit and explicit products is likely to lead to significant complexity and to be a source of both confusion and bugs.
-
-An example of a package containing explicit product definitions:
+An example of a package that defines two library products and one executable product:
 
 ```
 let package = Package(
-    name: "Hello",
+    name: "MyServer",
     targets: [
-        Target(name: "Foo"),
-        Target(name: "Bar"),
-        Target(name: "Baz", dependencies: ["Foo", "Bar"]),
-        Target(name: "Exe", dependencies: ["Foo"])
+        Target(name: "Utils"),
+        Target(name: "HTTP", dependencies: ["Utils"]),
+        Target(name: "ClientAPI", dependencies: ["HTTP", "Utils"]),
+        Target(name: "ServerAPI", dependencies: ["HTTP"]),
+        Target(name: "ServerDaemon", dependencies: ["ServerAPI"])
     ],
     products: [
-        Library(name: "Lib1", type: .static, targets: ["Bar"]),
-        Library(name: "Lib2", type: .dynamic, targets: ["Baz"]),
-        Executable(name: "Exe", targets: ["Exe"]),
+        Library(name: "ClientLib", type: .static, targets: ["ClientAPI"]),
+        Library(name: "ServerLib", type: .dynamic, targets: ["ServerAPI"]),
+        Executable(name: "myserver", targets: ["ServerDaemon"]),
     ]
 )
 ```
 
-The initial types of products that can be defined are executables and libraries.  Libraries can be declared as static or dynamic, but when possible, the specific type of library should be omitted -- in this case, the build system will choose the most appropriate type to build based on the context in which the product will be used (depedning on the type of client, the platform, etc).
+The initial types of products that can be defined are executables and libraries.  Libraries can be declared as either static or dynamic, but when possible, the specific type of library should be left unspecified, letting the build system chooses a suitable default for the platform.
```

Comment:
I see, so we will start out with a default of one or the other, but tell users not to specify it and then later it will change to the "auto" type? Sounds like a good plan.